### DISCUSSION TITLE
Receiving 504 errors in vercel logs. Using Stripe and MongoDB.

### DISCUSSION THREAD

**Discussion by bwestwood11** [vercel/next.js#52301] [Help]
### Summary

My localhost:3000 is working perfectly fine, but when I go into production then it doesn't work and I get a 504 error because it's taking more than 10 seconds for the request. 

I did add all environment variables inside of Vercel and checked all three environments as well. 

Tried checking the logs and don't see the error message. I am pretty stuck and need some help. I am using Stripe API and MongoDB

**Comment by icyJoseph** [comment]
A few more things to double check:

- Bundle size, https://vercel.com/docs/concepts/functions/serverless-functions#bundling-serverless-functions
- Unfinished async work, less likely but maybe in production the data/responses you use are failing, or never resolving?

**Comment by hassan-scholarden** [comment]
I am also getting this same error, I have add a stripe subscription api in nextjs api route, I also thougth that this issue is related to stripe, but in my case my all API routes are not working in production environment, I am stuck in this because it there is no proper error message, 
I am just receiving a 504 status code
